### ISSUE TITLE
pecl reinstall package every puppet execution

### ISSUE THREAD

**Issue by stupied4ever** [mayflower/puppet-php#120]
Every execution puppet reinstalls the pecl modules (probably because the pecl upgrade). This is a big problem as it forces the fpm to reload (causing backend connection problem).

My hiera:
```yaml
php::extensions:
  apc:
    provider: pecl
  pdo_mysql:
    provider: pecl
```

Agent output:
```
Notice: /Stage[main]/Php/Php::Extension[pdo_mysql]/Package[pecl-pdo_mysql]/ensure: created
Notice: /Stage[main]/Php/Php::Extension[apc]/Package[pecl-apc]/ensure: created
```

Exist any workaround?

**Comment by stupied4ever** [comment]
@globin  Do you know if exists any workaround? Otherwise i will need to put my pecl installation on my 
manifest (really bad).

**Comment by fpletz** [comment]
The pecl package provider is a little bit broken unfortunately. We have already looked at it a few times but we don't currently know of any workaround. :disappointed:

**Comment by prachetasp** [comment]
@stupied4ever check out #144

**Comment by jsok** [comment]
I'm seeing the same issue, even worse is that in `fpm.pp` it has `::Php::Extension <| |> ~> Service[$service_name]` means FPM gets restarted on every puppet run.

**Comment by fpletz** [comment]
This should be fixed with #144.
